### ISSUE TITLE
Allow visitors to search and filter documents and law library content

### ISSUE THREAD

**Issue by joeg8612** [massgov/feature-requests#212]
**Is your feature request related to a problem? Please describe.**
There are often collections of documents and pages that constituents would like to search within or filter. The standard site search is too broad. There should be a focused way to search and filter a collection from a page such as an info detail page.

**Describe the solution you'd like**
Documents and other content types (advisory, decision, rules of court, regulation, exec order) currently have a "type" field and a "labels" field.  We can consistently make sure that the type field is in metadata and we can also add the labels to metadata.  We then can create a component that allows visitors to search for content with criteria such as content type, organization, subtype and labels. The author could predefine the query parameters which could match groups of documents or pages.  There may need to be custom fields displayed in the results rather than just a simple link.  Requirements analysis needed.

Possible collection include:
[Audits](https://www.mass.gov/lists/all-audit-reports) 
[Inspector General reports](https://www.mass.gov/service-details/oig-2019-reports-and-recommendations)
[DOB enforcement actions](https://www.mass.gov/info-details/enforcement-actions-issued-by-the-division-of-banks#2019-)
[DOI public hearing decisions](https://www.mass.gov/lists/division-of-insurance-public-hearing-decisions)
[DOI regulatory bulletins](https://www.mass.gov/lists/doi-bulletins)
[DALA General Jurisdiction Decisions](https://www.mass.gov/general-jurisdiction-decisions)
[MCAD Hearing decisions](https://www.mass.gov/lists/mcad-hearing-decisions)
[Ethics decisions / rulings](https://www.mass.gov/read-laws-regulations-legal-opinions-and-decisions)
[ATB Findings and reports](https://www.mass.gov/lists/atb-findings-of-fact-and-reports)
[Civil service commission decisions](https://www.mass.gov/lists/civil-service-commission-decisions-2017-present)
[DOR Technical information releases](https://www.mass.gov/lists/dor-technical-information-releases)
[PERAC Memos](https://www.mass.gov/perac-memos)

**Additional context**
An example of other sites that handle this well:
https://www.sao.wa.gov/reports-data/audit-reports/

This was requested by the State Auditor's office.

**Reference**
n/a

**Comment by MikeWessler** [comment]
This is one of the features that is most requested by individuals seeking information on our audits and other reports. While the existing search functionality on Mass.gov is helpful, it is often too broad and cluttered because press releases and other content types that may contain particular words are included as well. Additionally, as we've moved our audits to an html format (which is undoubtedly a positive thing), the clutter has become worse, as users now also have to sort through various info details pages as well. Giving visitors to our site the flexibility to search audits by year/secretariat/topic and other customizable features would be incredibly powerful.

**Comment by Lchiong** [comment]
Agreed, I just had an Office Hours session with Kevin on this very topic. Our leadership has expressed interest in expanded functionality with filtered searches, particularly when it comes to our bulletins, advisories and reports. More customizable searches would be a great resource for our constituents. (OIG)

**Comment by swunderlee** [comment]
MassDOT also has a need for a document search, we like this example on the MN DOT website:
https://researchprojects.dot.state.mn.us/projectpages/pages/homepage.jsf;jsessionid=54D508C61B3438C9995993D4BBF6A5D3

**Comment by joeg8612** [comment]
This is near release. The feature will be called "Collections".  If you are interested in trying this before release, reach out to Joe.

**Comment by joeg8612** [comment]
Collection functionality has been released which will help support this need. This does not support full text search of pages in the collection. Search includes only the title and short description of the items in the collection.
